tools/Postfix_relay/mailwatch_milter_relay.php

```diff
@@ -50,7 +50,7 @@ function doit($input)
     while ($line = fgets($fp, 2096)) {
         if (preg_match('/^.*postfix\/cleanup.*: (\S+): message-id=(\S+)$/', $line, $explode)) {
             $smtp_id = $explode[1];
-            $message_id = $explode[2];
+            $message_id = preg_replace("'", "''", $explode[2]);
```

Comment:
I think that a `safe_value()` on both $smtp_id and $message_id must be used, because these 2 values are used in SQL query as is, and are prone to sql inject attack

Reply:
```php
$smtp_id = safe_value($explode[1]);
$message_id = safe_value($explode[2]);
```

Reply:
That makes more sense.